### PR TITLE
refactor(core): Move `instanceType` to `InstanceSettings` (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
[Context](https://n8nio.slack.com/archives/C069HS026UF/p1724744752553359)

## Review / Merge checklist

- [x] PR title and summary are descriptive

### CODE REVIEW

File: packages/cli/src/services/orchestration/webhook/handle-command-message-webhook.ts [hunk 1]
```diff
@@ -1,15 +1,17 @@
+import { Container } from 'typedi';
+import { InstanceSettings } from 'n8n-core';
+
 import { ExternalSecretsManager } from '@/external-secrets/external-secrets-manager.ee';
+import { Logger } from '@/logger';
 import { License } from '@/license';
 import { MessageEventBus } from '@/eventbus/message-event-bus/message-event-bus';
-import Container from 'typedi';
-import { Logger } from 'winston';
```
Comment (netroy) [inline packages/cli/src/services/orchestration/webhook/handle-command-message-webhook.ts:5]: not sure how this was even working considering that `winston.Logger` doesn't exist on the DI container.

File: packages/core/src/InstanceSettings.ts [hunk 2]
```diff
@@ -66,6 +68,13 @@ export class InstanceSettings {
 		this.instanceRole = 'follower';
 	}
 
+	// TODO: read from process.env instead, and make it readonly
+	instanceType?: InstanceType;
```
Comment (netroy) [inline packages/core/src/InstanceSettings.ts:72]: should we read this from `process.argv` in the constructor instead?

File: packages/cli/src/interfaces.ts [hunk 3]
```diff
@@ -630,5 +630,3 @@ export abstract class SecretsProvider {
 	abstract hasSecret(name: string): boolean;
 	abstract getSecretNames(): string[];
 }
-
-export type N8nInstanceType = 'main' | 'webhook' | 'worker';
```
Comment (tomi) [inline packages/cli/src/interfaces.ts:634]: 🎉

**Review by tomi — DISMISSED** [review]
Nice, much clearer this way 💟 Would be nice have getters for the different instance types

**Review by tomi — APPROVED** [review]
Looks good 🚀 I still feel having separate getters like `isMainInstance` etc would make the comparisons cleaner

**Comment by netroy** [comment]
> separate getters like `isMainInstance` etc would make the comparisons cleaner

Agreed. We can do this in a separate PR 🙏🏽

**Comment by janober** [comment]
Got released with [`n8n@1.60.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.60.0)

**Pull request merged**